src/main/scala/au/com/onegeek/sbtdotenv/SbtDotenv.scala

```diff
@@ -82,26 +82,49 @@ object SbtDotenv extends AutoPlugin {
       None
     } else {
       val source = Source.fromFile(file)
-      val result = source.getLines.flatMap(parseLine).toMap
+      val result = parse(source)
       source.close
       Some(result)
     }
   }
 
-  def isValidLine(line: String): Boolean = line.matches("^[a-zA-Z_]+[a-zA-Z0-9_]*=.*")
+  private val LINE_REGEX =
+    """(?xms)
+       (?:^|\A)                  # start of line
+       \s*                       # leading whitespace
+       (?:export\s+)?            # export (optional)
+       ([a-zA-Z_]+[a-zA-Z0-9_]*) # variable name (captured)
+       (?:\s*=\s*?)              # assignment with whitespace
+       (                         # start variable value (captured)
+         '(?:\\'|[^'])*'           # single quoted variable
+         |                         # or
+         "(?:\\"|[^"])*"           # double quoted variable
+         |                         # or
+         [^\#\r\n]+                # unquoted variable
+       )                         # end variable value (captured)
+       \s*                       # trailing whitespace
+       (?:                       # start trailing comment (optional)
+         \#                        # begin comment
+         (?:(?!$).)*               # any character up to end-of-line
+       )?                        # end trailing comment (optional)
+       (?:$|\z)                  # end of line
+    """.r
```

Comment:
do you have a reference to this regex somewhere? Is it used in other projects?

Reply:
There's a very similar one in the ruby dotenv https://github.com/bkeepers/dotenv/blob/master/lib/dotenv/parser.rb#L14

The main difference is that the SbtDotEnv one runs in multiline mode to make supporting multiline variables more straightforward which means tweaking the trailing comment capture to capture to the end of the current line and not forever.